flaml/tune/searcher/flow2.py

```diff
@@ -364,8 +364,8 @@ def update_fbest(
         for k_metric in self.lexico_objectives["metrics"]:
             k_values = np.array(self._histories[k_metric])
             self._f_best[k_metric] = np.min(k_values.take(feasible_index))
-            feasible_index_prior = np.where(
-                k_values
+            feasible_index_filter = np.where(
+                k_values.take(feasible_index)
```

Comment:
`k_values.take(feasible_index)` is calculated twice. Use a var to calculate only once.